Buffs androids by giving them many things you would expect from a robot, and gives nanites.

Uristthedorf opened the pull request:
<!-- If this is your first PR, or not, take the time to read our CONTRIBUTING.md file! You can see it here: https://github.com/yogstation13/Yogstation/blob/master/.github/CONTRIBUTING.md
You can remove all headers (Document the changes, Spriting and Wiki documentation) if there is no wiki documentation required but you must still explain what the pr is and why it needs to be added to the game. Directors+ Are immune from this rule in exceptional circumstances. -->

# Document the changes in your pull request

As of now, the android species is bare bones and very weak compared to augmented humans. This PR gives androids things you would expect from a robot. In addition this gives any mob that is not a silicon the ability to use nanites.
A list of changes to androids would be:

- Having cybernetic organs instead of all organic except for tongue.
- Immunity to toxin and clone damage.
- NOZOMBIE, NOHUSK, TRAIT_COLDBLOODED, TRAIT_NOCLONE, TRAIT_TOXIMMUNE, TRAIT_MEDICALIGNORE traits.
- Iron/plasteel when you skin/gib them.
- The "SYNTHETIC" regent processing tag.
- A lot of mostly cosmetic changes I stole from IPC to make androids feel/sound more robotic.

~~The reason why I gave all mobs that are robotic and humanoid nanites is because it seemed less snowflakey to do that than to add another specific species check to nanites. A list of species that get nanites from this I believe would be: Androids, synthetics, clock golems.~~

I feel like this brings androids closer balance wise to augged humans. Augged humans still have access to experimental surgeries for things like halving their stun mod or becoming immune to shocks, and can also get mutations. (And cyborg limbs make humans immune to space I think, so the temp/pressure immunity androids get doesn't do too much to make them better than augged humans.)

I have tested my code.

I considered giving EMP resistance/immunity to androids (or only chaplain androids) to give them an advantage over augged humans because augged humans get stuff like surgery and meth while androids can't. But that is probably best left for another PR. (Surely it wouldn't be more OP than a jellyperson getting space adapt to remove their cold weakness while still keeping the 0.5 burn mod and slime powers. Who can then get more mutations/augs whatnot.)

# Wiki Documentation

You could add the buffs I gave to androids at https://wiki.yogstation.net/wiki/Guide_to_races.

# Changelog

<!-- Edit the changelog below to reflect the changes made by this PR, even if the changes are minor - required for every PR that has player-facing changes.
If you add a name after the ':cl:', that name will be used in the changelog. Leave it empty to use your GitHub name. -->

:cl:  
rscadd: Buffs androids by giving them many things you would expect from a robot, and cosmetic changes.
tweak: Lets anything that isn't a silicon get nanites.
/:cl:

Uristthedorf committed:
Update android.dm

Make "holy androids" EMP proof

Uristthedorf committed:
Update rites.dm

Makes chaplain rite make holy androids

Uristthedorf committed:
Update nanites.dm

Allows anything that is both humanoid and robotic to get nanites, including synths and androids.

Uristthedorf committed:
Update android.dm

Adds NOZOMBIE and NOHUSK to androids because they're robots and should have these.

Uristthedorf committed:
Update android.dm

Adds more traits to androids. (Why could they take clone and toxin damage?)

Uristthedorf committed:
Cybernetic organs for androids

Gives androids cybernetic organs. (They used to be all organic except for the tongue.)

Uristthedorf committed:
Update android.dm

Removed TRAIT_NOCRITDAMAGE, because they shouldn't get stuck in crit forever.

Uristthedorf committed:
Update android.dm

Makes androids immune to toxin and cloning damage because they are robots.

Uristthedorf committed:
Merge remote-tracking branch 'upstream/master' into holyAndroid

Uristthedorf committed:
Update android.dm

Copies a bunch of stuff from IPC to make them more robotic, mostly cosmetic except for PROCESS_SYNTHETIC regent processing, and the meat/skin being added.

Uristthedorf committed:
Update android.dm

Increases android limb protection to match that of augged limbs.

Uristthedorf committed:
Removes holy androids

Making chaplain androids EMP immune should probably be a separate PR.

Uristthedorf committed:
Update nanites.dm

typo fix

Uristthedorf committed:
Update nanites.dm

Syntax fix

Uristthedorf committed:
Update android.dm

Fixes var

Uristthedorf committed:
Update android.dm

Fixes limb damage reduction

Uristthedorf committed:
Finishes making nanites available to androids.

Forgot to touch the other files.

Uristthedorf committed:
Update android.dm

Adds robotic brute/burn messages.

Uristthedorf committed:
Lets anything that isn't a silicon get nanites.

Uristthedorf committed:
Revert "Update android.dm"

This reverts commit f5fdc8dc813d437b0fec56e89dec8a59b508642d.

ynot01 reviewed (dismissed):
Balance Council: Remove the limb damage reduction, the species is strong as is

Moltijoe reviewed (approved):
can't find any problems with the code, just make the change ynot requested and i think it's good to go

Uristthedorf committed:
Removed limb damage reduction.

Bibby0110 reviewed (changes requested):
Add a species trait to disallow nanites

Uristthedorf committed:
Adds NONANITES trait

I was told that you should check if a mob is a carbon before checking carbon traits. (I have not tested this but it compiles.)

Uristthedorf committed:
Update code/datums/components/nanites.dm

Co-authored-by: TheGamerdk <5618080+TheGamerdk@users.noreply.github.com>

Uristthedorf committed:
Update code/datums/components/nanites.dm

Co-authored-by: TheGamerdk <5618080+TheGamerdk@users.noreply.github.com>
